### PR TITLE
mds: send scrub status to ceph-mgr only when scrub is running

### PR DESCRIPTION
**Pull request by vshankar**

### CODE REVIEW

File: qa/tasks/cephfs/test_scrub_checks.py [hunk 1]
```diff
@@ -30,22 +30,48 @@ def _resume_scrub(self, expected):
...
     def _get_scrub_status(self):
         return self.fs.rank_tell(["scrub", "status"])
-    def _check_task_status(self, expected_status):
-        task_status = self.fs.get_task_status("scrub status")
-        active = self.fs.get_active_names()
-        log.debug("current active={0}".format(active))
-        self.assertTrue(task_status[active[0]].startswith(expected_status))
+    def _check_task_status(self, expected_status, timo=120):
+        check = 0
+        while check < timo:
```
Comment (batrick) [inline qa/tasks/cephfs/test_scrub_checks.py:35]: check out `safe_while` elsewhere in the qa suite / teuthology

File: qa/tasks/cephfs/test_scrub_checks.py [hunk 2]
```diff
@@ -30,22 +30,48 @@ def _resume_scrub(self, expected):
...
+            try:
+                if task_status[active[0]].startswith(expected_status):
+                    return True
+            except KeyError:
+                pass
+            check += 1
+            time.sleep(1)
+        return False
+
+    def _check_task_status_na(self, timo=120):
```
Comment (batrick) [inline qa/tasks/cephfs/test_scrub_checks.py:48]: `na` is?

File: qa/tasks/cephfs/test_scrub_checks.py [hunk 3]
```diff
@@ -30,22 +30,48 @@ def _resume_scrub(self, expected):
...
+
+    def create_scrub_data(self, test_dir):
+        self.mount_a.run_shell(["mkdir", test_dir])
+        for i in range(32):
+            dirname = "dir.{0}".format(i)
+            dirpath = os.path.join(test_dir, dirname)
+            self.mount_a.run_shell(["mkdir", dirpath])
+            for j in range(32):
+                filename = "file.{0}".format(j)
+                self.mount_a.write_n_mb(os.path.join(dirpath, filename), 1)
```
Comment (batrick) [inline qa/tasks/cephfs/test_scrub_checks.py:68]: Also check out `self.run_shell_payload`. I think we should start moving away from these awkward numerous calls to `run_shell` when a trivial monolithic shell script will do.

File: qa/tasks/cephfs/test_scrub_checks.py (hunk 1 above)
Comment (vshankar) [reply qa/tasks/cephfs/test_scrub_checks.py:35]: ACK -- nice to know

File: qa/tasks/cephfs/test_scrub_checks.py (hunk 2 above)
Comment (vshankar) [reply qa/tasks/cephfs/test_scrub_checks.py:48]: "not available" -- will make that clear.

File: qa/tasks/cephfs/test_scrub_checks.py (hunk 3 above)
Comment (vshankar) [reply qa/tasks/cephfs/test_scrub_checks.py:68]: ACK

**Comment by batrick** [comment]
jenkins retest this

**Comment by batrick** [comment]
retest this please

File: qa/tasks/cephfs/test_scrub_checks.py [hunk 4]
```diff
@@ -56,19 +81,20 @@ def test_scrub_abort(self):
...
-        time.sleep(10)
-        self._check_task_status("idle")
+        checked = self._check_task_status_na()
+        self.assertTrue(checked)
 
     def test_scrub_pause_and_resume(self):
         test_dir = "scrub_control_test_path"
         abs_test_path = "/{0}".format(test_dir)
 
+        success_validator = lambda j, r: TestScrubChecks.json_validator(j, r, "return_code", 0)
```
Comment (batrick) [inline qa/tasks/cephfs/test_scrub_checks.py:91]: `./tasks/cephfs/test_scrub_checks.py:91:9: F841 local variable 'success_validator' is assigned to but never used`

Comment (vshankar) [reply qa/tasks/cephfs/test_scrub_checks.py:91]: ack

**Comment by vshankar** [comment]
fixed and updated

**Comment by votdev** [comment]
Why has this PR been merged, although the Dashboard API tests are failing? We now have a regression: https://tracker.ceph.com/issues/46495

**Comment by smithfarm** [comment]
This PR introduced a regression - https://tracker.ceph.com/issues/46495 - which is being fixed by https://github.com/ceph/ceph/pull/36091

**Comment by batrick** [comment]
> This PR introduced a regression - https://tracker.ceph.com/issues/46495 - which is being fixed by #36091
>
> (Since there is no tracker associated with this PR, I'm guessing that it's not a candidate for backport, but just in case...)

https://tracker.ceph.com/issues/45349 ?

**Comment by smithfarm** [comment]
> https://tracker.ceph.com/issues/45349 ?

Yeah - added that to the PR description in the meantime.

**Comment by vshankar** [comment]
> Why has this PR been merged, although the Dashboard API tests are failing? We now have a regression: https://tracker.ceph.com/issues/46495
> Please contact the Dashboard team to clarify such issues the next time before merging.

I think there were other failures which probably masked this regression...

**Comment by tchaikov** [comment]
> > Why has this PR been merged, although the Dashboard API tests are failing? We now have a regression: https://tracker.ceph.com/issues/46495
> > Please contact the Dashboard team to clarify such issues the next time before merging.
> 
> I think there were other failures which probably masked this regression...

please see https://jenkins.ceph.com/job/ceph-dashboard-pr-backend/3522/

which has the first occurrence of https://tracker.ceph.com/issues/46495
```
Python Unit-test Failure

Failures: 1

test_get_export (tasks.mgr.dashboard.test_ganesha.GaneshaTest)
2020-07-14 11:45:43,994.994 INFO:__main__:----------------------------------------------------------------------
2020-07-14 11:45:43,994.994 INFO:__main__:Traceback (most recent call last):
2020-07-14 11:45:43,994.994 INFO:__main__:  File "/home/jenkins-build/build/workspace/ceph-dashboard-pr-backend/qa/tasks/mgr/dashboard/test_ganesha.py", line 148, in test_get_export
2020-07-14 11:45:43,994.994 INFO:__main__:    self.assertDictEqual(export2, data2)
2020-07-14 11:45:43,994.994 INFO:__main__:AssertionError: {'status': '404 Not Found', 'detail': 'Noth[76 chars]85d'} != {'detail': "Cannot create this export. FSAL[181 chars]2'}}}
2020-07-14 11:45:43,994.994 INFO:__main__:- {'detail': 'Nothing matches the given URI',
2020-07-14 11:45:43,994.994 INFO:__main__:-  'request_id': '4b8fbb53-e593-4ae9-82ca-cbec8c2dd85d',
2020-07-14 11:45:43,995.995 INFO:__main__:-  'status': '404 Not Found'}
2020-07-14 11:45:43,995.995 INFO:__main__:+ {'component': 'nfs',
2020-07-14 11:45:43,995.995 INFO:__main__:+  'detail': "Cannot create this export. FSAL 'RGW' cannot be managed by the "
2020-07-14 11:45:43,995.995 INFO:__main__:+            'dashboard.',
2020-07-14 11:45:43,995.995 INFO:__main__:+  'status': 400,
2020-07-14 11:45:43,995.995 INFO:__main__:+  'task': {'metadata': {'cluster_id': 'cluster2',
2020-07-14 11:45:43,995.995 INFO:__main__:+                        'fsal': 'RGW',
2020-07-14 11:45:43,995.995 INFO:__main__:+                        'path': 'mybucket'},
2020-07-14 11:45:43,995.995 INFO:__main__:+           'name': 'nfs/create'}}
2020-07-14 11:45:43,995.995 INFO:__main__:
2020-07-14 11:45:43,995.995 INFO:__main__:----------------------------------------------------------------------
2020-07-14 11:45:43,996.996 INFO:__main__:Ran 63 tests in 670.669s
2020-07-14 11:45:43,996.996 INFO:__main__:
2020-07-14 11:45:43,996.996 INFO:__main__:
```

**Pull request merged**